src/sql-parser/src/parser.rs

```diff
@@ -1659,9 +1659,46 @@ impl<'a> Parser<'a> {
             envelope,
             if_not_exists,
             materialized,
+            key_constraint,
         }))
     }
 
+    /// Parses the column section of a CREATE SOURCE statement which can be
+    /// empty, a list of columns with an optional key constraint, or a key
+    /// constraint.
+    fn parse_source_columns(&mut self) -> Result<(Vec<Ident>, Option<KeyConstraint>), ParserError> {
+        if self.consume_token(&Token::LParen) {
+            let mut columns = vec![];
+            let mut key_constraint;
+            loop {
+                key_constraint = self.parse_key_constraint()?;
+                if key_constraint.is_some() {
+                    break;
```

Comment:
This would only allow the key constraint in the last position. We could be more lax and just allow it anywhere.

Reply:
I assumed that's what we wanted here, but after double checking table constraints in Postgres, I think that would make sense (it would also allow us to re-use the existing `parse_comma_separated`).

Reply:
Done in 1931e9516 (but re-using `parse_comma_separated` was not super convenient, so I didn't).